core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -952,14 +952,24 @@ private void copyLogSegment(UnifiedLog log, LogSegment segment, RemoteLogSegment
             brokerTopicStats.topicStats(log.topicPartition().topic()).remoteCopyRequestRate().mark();
             brokerTopicStats.allTopicsStats().remoteCopyRequestRate().mark();
             Optional<CustomMetadata> customMetadata = Optional.empty();
+            
             try {
                 customMetadata = remoteLogStorageManager.copyLogSegmentData(copySegmentStartedRlsm, segmentData);
             } catch (RemoteStorageException e) {
+                logger.warn("Cleaning after failing to copy segment {}", copySegmentStartedRlsm.remoteLogSegmentId());
                 try {
-                    remoteLogStorageManager.deleteLogSegmentData(copySegmentStartedRlsm);
-                    logger.info("Successfully cleaned segment {} after failing to copy segment", segmentId);
+                    deleteRemoteLogSegment(
```

Comment:
Se should also update for L991 that `"Custom metadata size exceeds configured limit` case.

Reply:
Yep, just noticed.